### ISSUE TITLE
Using Separate References and Bypassing Gender Inference in Fix, Call, Segment, etc. 

### ISSUE THREAD

**Issue by charliecurnin** [etal/cnvkit#241]
Hi Eric, 

Because of the difficulties I had in getting CNVkit to detect the gender of my samples correctly, I ended up making separate references for male and female samples. 

I ran `reference` over all female samples to create a female reference; I ran `reference ... --male-reference` over all male samples to create a male reference. 

Now that I have these separate references, though, I want to make sure what I'm doing is correct. 

- With `fix`, I should call all female samples only against the female reference, and the male samples against the male reference. I didn't see anything mentioning using a `-y` flag with `fix`—is that necessary when working with male samples? I haven't attempted it just yet, but I'm somewhat worried that CNVkit will consider some of the male samples female (and interpret `-y` just as indicating that the *reference* is male), and then call them as female against a male reference. Is this true? Is there a way to bypass gender inference in `fix`? Is gender inference even relevant to `fix`?
- The [page](http://cnvkit.readthedocs.io/en/stable/sex.html#reference-sex) on chromosomal sex says 

   >If you want copy number calls to be relative to a male reference with a single X chromosome but dipoid autosomes, use the -y option everywhere. 

  - Does that include `fix`? Or `segment`? Is gender even relevant for these?
  - With `call` I can see that `-y` should be used "if a male reference was used." But does that mean that even if I use `-y`, CNVkit will still try to guess the gender of my samples?—as opposed to recognizing that I would only call male samples against a male reference? I edited `reference.py`to have CNVkit consult lists of male and female samples before calling `is_xx()`, but I'm not sure where I would do this in `call.py`?

Basically, for any arbitrary command (namely, `fix`, `segment`, and `call`), if I'm calling female samples with a female reference, and male samples with a male reference, what do I need to do besides use `-y` with the male samples?

**Comment by etal** [comment]
If the input samples are all male, you want to use the option `-x male` as well as `-y` to ensure no inference is done. The `-x` option skips inference on the input files; the `-y` option just determines whether the reference profile should have an X ploidy of 1 or 2, and can be used with male or female samples, or a mix.

Any CNVkit command that detects chromosomal sex will have options `-x` and `-y`. These are: batch, reference, call, diagram, heatmap, gainloss, sex, export bed/vcf. For the other commands it doesn't matter.

If you use `-y` with `reference`, then you have to use it with all of the other commands that have the `-y` option, too. The `-x` option is optional everywhere, and if you forget to use it in one part of the pipeline, it will fall back to automatic inference (which tends to work better on .cnr and .cns files than on targetcoverage.cnn).

**Comment by charliecurnin** [comment]
Thank you so much. This is really helpful.
